app/main/lib/elasticsearch.py

```diff
@@ -86,7 +86,8 @@ def update_or_create_document(body, doc_id, index):
           result = es.update(
               id=doc_id,
               body={"doc": merge_contexts(body, found_doc)},
-              index=index
+              index=index,
+              retry_on_conflict=3
```

Comment:
OpenSearch documentation is here - https://opensearch.org/docs/1.2/opensearch/rest-api/document-apis/update-document/

Reply:
P.S. Agree that it would be great to try to test this locally by sending requests to Alegre's `similarity/text` endpoint quickly via `curl` with the same text and different contexts.

Reply:
I have not run this locally yet - I think we have to just let this roll out and see what the effect is. It's a well-established parameter we just aren't using right now.